exporters/trace/fluentforward/fluentforward.go

```diff
@@ -0,0 +1,199 @@
+	return func(o *options)	{
+		o.logger = logger
+	}
+}
+
+func WithSDK(config *sdktrace.Config) Option{
+	return func(o *options){
+		o.config = config
+	}
+}
+
+func InstallNewPipeline(ffurl, serviceName string, opts ...Option) error{
+	tp,err := NewExportPipeline(ffurl, serviceName, opts...)
+	if err!=nil{
+		return err
+	}
+	global.SetTracerProvider(tp)
+	return nil
+}
+
+func NewExportPipeline(ffurl, serviceName string, opts ...Option) (*sdktrace.TracerProvider, error){
+	exp, err := NewExporter(ffurl, serviceName, opts...)
+	if err!=nil{
+		return nil, err
+	}
+	tp := sdktrace.NewTracerProvider(sdktrace.WithBatcher(exp))
+	return tp, nil
+}
+
+func NewExporter(ffurl, serviceName string, opts ...Option) (*Exporter, error){
+	o := options{}
+	for _, opt := range opts{
+		opt(&o)
+	}
+
+	if ffurl == ""{
+		return nil, errors.New("Fluent instance url cannot be empty")
+	}
+
+	return &Exporter{
+		url: ffurl,
+		serviceName: serviceName,
+		o: o,
+	}, nil
+
+}
+
+func (e *Exporter) ExportSpans(ctx context.Context, sds []*export.SpanData) error{
+
+	ts := time.Now().Unix()
+
+	tcpAddr, err := net.ResolveTCPAddr("tcp", e.url)
+	check(err)
+	client, err := net.DialTCP("tcp", nil, tcpAddr)
+	e.client = client
+	check(err)
+
+	for _, span := range sds{
+		ffspan := ffSpan{
+			Tag: "span.test",
+			Ts: ts,
+		}
+		spans := spanData{}
+		spans.SpanContext = spanContext{
+			TraceId: span.SpanContext.TraceID.String(),
+			SpanId: span.SpanContext.SpanID.String(),
+			TraceFlags: span.SpanContext.TraceFlags,
+		}
+		spans.ParentSpanId = span.ParentSpanID.String()
+		spans.SpanKind = span.SpanKind
+		spans.Name = span.Name
+		spans.StatusMessage = span.StatusMessage
+		spans.StatusCode = span.StatusCode.String()
+		spans.StartTime = span.StartTime.UnixNano()
+		spans.EndTime = span.EndTime.UnixNano()
+		spans.DroppedAttributeCount = span.DroppedAttributeCount
+		spans.DroppedLinkCount = span.DroppedLinkCount
+		spans.DroppedMessageEventCount = span.DroppedMessageEventCount
+		
+		spans.Attrs = attributesToMap(span.Attributes)
+	
+		spans.Links = linksToSlice(span.Links)
+
+		ffspan.SpanData = spans
+
+		t, err := msgpack.Marshal(&ffspan)
+		check(err)
+
+		_, err = e.client.Write(t)
+		check(err)
+	}
+	return nil
+}
+
+func (e *Exporter) Shutdown(ctx context.Context) error{
+	e.client.Close()
+	return nil
+}
+
+func attributesToMap(attributes []label.KeyValue) map[label.Key]interface{}{
+		attrs := make(map[label.Key]interface{})
+		for _,v := range attributes{
+			attrs[v.Key] = v.Value.AsInterface()
+		}
+		return attrs
+}
+
+func linksToSlice(links []apitrace.Link) []link{
+	var l []link
+	for _, v := range links{
+		temp := link{
+			SpanContext: spanContext{
+				TraceId: v.SpanContext.TraceID.String(),
+				SpanId: v.SpanContext.SpanID.String(),
+				TraceFlags: v.SpanContext.TraceFlags,
+			},
+			Attrs: attributesToMap(v.Attributes),
+		}
+		l = append(l, temp)	
+	}
+	return l
+}
+
+func check(err error){
```

Comment:
why do you need this helper method? It is super scary to panic when export cannot happen